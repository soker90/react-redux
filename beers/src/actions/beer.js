import beerAction from '../constants/actions/beer'

export default {
  fetchBeer: () => ({type: beerAction.FETCH_BEER}),
  fetchBeerSuccess: beers => ({type: beerAction.FETCH_BEER_SUCCESS, beers}),
  addBeer: (name, graduation, price) => ({type: beerAction.ADD_BEER, name, graduation, price}),
  addBeerSuccess: beer => ({type: beerAction.ADD_BEER_SUCCESS, beer}),
  removeBeer: beerId => ({type: beerAction.REMOVE_BEER, beerId}),
  removeBeerSuccess: beerId => ({type: beerAction.REMOVE_BEER_SUCCESS, beerId}),
  beerError: error => ({type: beerAction.BEER_ERROR, error})
}
