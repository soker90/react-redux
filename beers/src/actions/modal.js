import modalAction from '../constants/actions/modal'

export default {
  openModal: modalData => ({type: modalAction.OPEN_MODAL, modalData}),
  closeModal: () => ({type: modalAction.CLOSE_MODAL})
}
