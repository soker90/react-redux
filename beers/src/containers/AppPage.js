import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import modalActions from '../actions/modal'
import App from '../components/App'

export function mapStateToProps(state) {
  return {modal: {...state.modal}}
}

export function mapDispatchToProps(dispatch) {
  return {modalAction: bindActionCreators(modalActions, dispatch)}
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
