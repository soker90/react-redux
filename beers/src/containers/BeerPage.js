import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import modalActions from '../actions/modal'
import beerActions from '../actions/beer'
import Beer from '../components/Beer'

export function mapStateToProps (state) {
  return {...state.beers, ...state.modal}
}

export function mapDispatchToProps (dispatch) {
  return bindActionCreators({...beerActions, ...modalActions}, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Beer)
