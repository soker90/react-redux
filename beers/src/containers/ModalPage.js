import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import modalActions from '../actions/modal'
import Modal from '../components/Modal'

export function mapStateToProps (state) {
  return {...state.modal}
}

export function mapDispatchToProps (dispatch) {
  return bindActionCreators(modalActions, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Modal)
