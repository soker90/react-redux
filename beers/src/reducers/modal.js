import modalAction from '../constants/actions/modal'

export function initialState () {
  return {
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}
  }
}

function openModal(state, {modalData}) {
  return {
    ...state,
    open: true,
    title: modalData.title,
    body: modalData.body,
    callback: modalData.callback ? modalData.callback : null,
    extraParams: modalData.extraParams ? modalData.extraParams : {}
  }
}

function closeModal(state) {
  return {
    ...state,
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}

  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case modalAction.OPEN_MODAL:
      return openModal(state, action)
    case modalAction.CLOSE_MODAL:
      return closeModal(state, action)
    default:
      return state
  }
}
