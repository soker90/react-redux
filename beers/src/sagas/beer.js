import { call, put, takeEvery } from 'redux-saga/effects'
import addBeerService from '../services/addBeer'
import fetchBeerService from '../services/fetchBeer'
import removeBeerService from '../services/removeBeer'
import beerActions from '../actions/beer'
import beerActionTypes from '../constants/actions/beer'

export function * fetchBeer () {
  try {
    const response = yield call(fetchBeerService)
    yield put(beerActions.fetchBeerSuccess(response))
  } catch (error) {
    yield put(beerActions.beerError(error))
  }
}

export function * watchFetchBeer () {
  yield takeEvery(beerActionTypes.FETCH_BEER, fetchBeer)
}

export function * addBeer (beer) {
  try {
    const response = yield call(addBeerService, beer)
    yield put(beerActions.addBeerSuccess(response))
  } catch (error) {
    yield put(beerActions.beerError(error))
  }
}

export function * watchAddBeer () {
  yield takeEvery(beerActionTypes.ADD_BEER, addBeer)
}

export function * removeBeer (beerId) {
  try {
    yield call(removeBeerService, beerId)
    yield put(beerActions.removeBeerSuccess(beerId))
  } catch (error) {
    yield put(beerActions.beerError(error))
  }
}

export function * watchRemoveBeer () {
  yield takeEvery(beerActionTypes.REMOVE_BEER, removeBeer)
}
