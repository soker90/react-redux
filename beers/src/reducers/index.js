import { combineReducers } from 'redux'
import beers from './beer'
import modal from './modal'
import coins from './coin'
import { reducer as formReducer } from 'redux-form'

const rootReducer = combineReducers({
  beers,
  modal,
  coins,
  form: formReducer
})

export default rootReducer
