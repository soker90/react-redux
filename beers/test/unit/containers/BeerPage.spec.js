import {mapStateToProps, mapDispatchToProps} from '../../../src/containers/BeerPage'
import { initialState } from '../../../src/reducers/modal'
import { initialState as beersInitialState } from '../../../src/reducers/beer'

describe('Containers', () => {
  test('should map BeerPage state to props', () => {
    const state = {
      modal: initialState(),
      beers: beersInitialState(),
    }

    expect(mapStateToProps(state)).toMatchSnapshot()
  })

  test('should map BeerPage actions to props', () => {
    expect(mapDispatchToProps()).toMatchSnapshot()
  })
})
