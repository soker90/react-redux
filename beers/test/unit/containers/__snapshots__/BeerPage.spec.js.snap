exports[`Containers should map BeerPage actions to props 1`] = `
Object {
  "addBeer": [Function],
  "addBeerSuccess": [Function],
  "beerError": [Function],
  "closeModal": [Function],
  "fetchBeer": [Function],
  "fetchBeerSuccess": [Function],
  "openModal": [Function],
  "removeBeer": [Function],
  "removeBeerSuccess": [Function],
}
`;

exports[`Containers should map BeerPage state to props 1`] = `
Object {
  "beer": Object {},
  "beerList": Array [],
  "body": "",
  "callback": null,
  "error": null,
  "extraParams": Object {},
  "loading": false,
  "open": false,
  "title": "",
}
`;
